Fix nightly build, add extra PR testing

drdaz opened the pull request:
The nightly build is still broken, and our PR testing is insufficient to prevent these types of surprises.

This PR fixes the nightly, and adds testing to PRs so we can more easily see whether our changes will break builds on master, and should help keep master and it's tests in a more stable state.

drdaz commented:
The CocoaPods issue is odd. Linting fails on with the --use-library argument. The build output is noisy. A LOT of warnings. Here's where ParseFacebookUtils-Dynamic-iOS fails, with some context: 

```
    CompileC /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFieldOperation.o /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/Parse/Parse/Internal/FieldOperation/PFFieldOperation.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target 'Parse' from project 'Pods')
        cd /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods
        export LANG=en_US.US-ASCII
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -target x86_64-apple-ios8.0-simulator -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu11 -fobjc-arc -fobjc-weak -fmodules -fmodules-cache-path=/Users/drdaz/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/drdaz/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -Os -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -DPOD_CONFIGURATION_RELEASE=1 -DCOCOAPODS=1 -DNS_BLOCK_ASSERTIONS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.4.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -g -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wno-semicolon-before-method-body -Wunguarded-availability -fobjc-abi-version=2 -fobjc-legacy-dispatch -iquote /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-generated-files.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-own-target-headers.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-all-target-headers.hmap -iquote /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-project-headers.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Products/Release-iphonesimulator/Parse/include -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private/Parse -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/Bolts -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/FBSDKCoreKit -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/FBSDKLoginKit -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/Parse -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources-normal/x86_64 -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources/x86_64 -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources -F/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Products/Release-iphonesimulator/Parse -Wincomplete-umbrella -include /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Target\ Support\ Files/Parse/Parse-prefix.pch -MMD -MT dependencies -MF /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFieldOperation.d --serialize-diagnostics /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFieldOperation.dia -c /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/Parse/Parse/Internal/FieldOperation/PFFieldOperation.m -o /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFieldOperation.o
    warning: Could not read serialized diagnostics file: Cannot Load File: Failed to open diagnostics file (in target 'Parse' from project 'Pods')
    
    CompileC /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookUtils.o /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/ParseFacebookUtils/ParseFacebookUtils/PFFacebookUtils.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target 'Parse' from project 'Pods')
        cd /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods
        export LANG=en_US.US-ASCII
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -target x86_64-apple-ios8.0-simulator -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu11 -fobjc-arc -fobjc-weak -fmodules -fmodules-cache-path=/Users/drdaz/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/drdaz/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -Os -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -DPOD_CONFIGURATION_RELEASE=1 -DCOCOAPODS=1 -DNS_BLOCK_ASSERTIONS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.4.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -g -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wno-semicolon-before-method-body -Wunguarded-availability -fobjc-abi-version=2 -fobjc-legacy-dispatch -iquote /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-generated-files.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-own-target-headers.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-all-target-headers.hmap -iquote /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-project-headers.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Products/Release-iphonesimulator/Parse/include -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private/Parse -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/Bolts -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/FBSDKCoreKit -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/FBSDKLoginKit -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/Parse -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources-normal/x86_64 -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources/x86_64 -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources -F/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Products/Release-iphonesimulator/Parse -Wincomplete-umbrella -include /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Target\ Support\ Files/Parse/Parse-prefix.pch -MMD -MT dependencies -MF /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookUtils.d --serialize-diagnostics /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookUtils.dia -c /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/ParseFacebookUtils/ParseFacebookUtils/PFFacebookUtils.m -o /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookUtils.o
    warning: Could not read serialized diagnostics file: Cannot Load File: Failed to open diagnostics file (in target 'Parse' from project 'Pods')
    
    CompileC /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookPrivateUtilities.o /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/ParseFacebookUtils/ParseFacebookUtils/Internal/PFFacebookPrivateUtilities.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target 'Parse' from project 'Pods')
        cd /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods
        export LANG=en_US.US-ASCII
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -target x86_64-apple-ios8.0-simulator -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu11 -fobjc-arc -fobjc-weak -fmodules -fmodules-cache-path=/Users/drdaz/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/drdaz/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -Os -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -DPOD_CONFIGURATION_RELEASE=1 -DCOCOAPODS=1 -DNS_BLOCK_ASSERTIONS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.4.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -g -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wno-semicolon-before-method-body -Wunguarded-availability -fobjc-abi-version=2 -fobjc-legacy-dispatch -iquote /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-generated-files.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-own-target-headers.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-all-target-headers.hmap -iquote /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-project-headers.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Products/Release-iphonesimulator/Parse/include -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private/Parse -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/Bolts -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/FBSDKCoreKit -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/FBSDKLoginKit -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/Parse -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources-normal/x86_64 -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources/x86_64 -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources -F/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Products/Release-iphonesimulator/Parse -Wincomplete-umbrella -include /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Target\ Support\ Files/Parse/Parse-prefix.pch -MMD -MT dependencies -MF /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookPrivateUtilities.d --serialize-diagnostics /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookPrivateUtilities.dia -c /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/ParseFacebookUtils/ParseFacebookUtils/Internal/PFFacebookPrivateUtilities.m -o /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookPrivateUtilities.o
    warning: Could not read serialized diagnostics file: Cannot Load File: Failed to open diagnostics file (in target 'Parse' from project 'Pods')
    
    CompileC /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookMobileAuthenticationProvider.o /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/ParseFacebookUtils/ParseFacebookUtils/Internal/AuthenticationProvider/iOS/PFFacebookMobileAuthenticationProvider.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target 'Parse' from project 'Pods')
        cd /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods
        export LANG=en_US.US-ASCII
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -target x86_64-apple-ios8.0-simulator -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu11 -fobjc-arc -fobjc-weak -fmodules -fmodules-cache-path=/Users/drdaz/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/drdaz/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -Os -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -DPOD_CONFIGURATION_RELEASE=1 -DCOCOAPODS=1 -DNS_BLOCK_ASSERTIONS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.4.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -g -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wno-semicolon-before-method-body -Wunguarded-availability -fobjc-abi-version=2 -fobjc-legacy-dispatch -iquote /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-generated-files.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-own-target-headers.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-all-target-headers.hmap -iquote /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-project-headers.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Products/Release-iphonesimulator/Parse/include -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private/Parse -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/Bolts -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/FBSDKCoreKit -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/FBSDKLoginKit -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/Parse -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources-normal/x86_64 -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources/x86_64 -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources -F/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Products/Release-iphonesimulator/Parse -Wincomplete-umbrella -include /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Target\ Support\ Files/Parse/Parse-prefix.pch -MMD -MT dependencies -MF /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookMobileAuthenticationProvider.d --serialize-diagnostics /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookMobileAuthenticationProvider.dia -c /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/ParseFacebookUtils/ParseFacebookUtils/Internal/AuthenticationProvider/iOS/PFFacebookMobileAuthenticationProvider.m -o /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookMobileAuthenticationProvider.o
    In file included from /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/ParseFacebookUtils/ParseFacebookUtils/Internal/AuthenticationProvider/iOS/PFFacebookMobileAuthenticationProvider.m:17:
    In file included from /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private/FBSDKLoginKit/FBSDKLoginKit.h:27:
    /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private/FBSDKLoginKit/FBSDKLoginButton.h:36:9: fatal error: module 'FBSDKCoreKit' not found
    @import FBSDKCoreKit;
     ~~~~~~~^~~~~~~~~~~~
    1 error generated.
    
    CompileC /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookAuthenticationProvider.o /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/ParseFacebookUtils/ParseFacebookUtils/Internal/AuthenticationProvider/PFFacebookAuthenticationProvider.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target 'Parse' from project 'Pods')
        cd /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods
        export LANG=en_US.US-ASCII
        /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -target x86_64-apple-ios8.0-simulator -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu11 -fobjc-arc -fobjc-weak -fmodules -fmodules-cache-path=/Users/drdaz/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/drdaz/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -Os -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -DPOD_CONFIGURATION_RELEASE=1 -DCOCOAPODS=1 -DNS_BLOCK_ASSERTIONS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.4.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -g -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wno-semicolon-before-method-body -Wunguarded-availability -fobjc-abi-version=2 -fobjc-legacy-dispatch -iquote /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-generated-files.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-own-target-headers.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-all-target-headers.hmap -iquote /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Parse-project-headers.hmap -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Products/Release-iphonesimulator/Parse/include -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Private/Parse -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/Bolts -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/FBSDKCoreKit -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/FBSDKLoginKit -I/var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Headers/Public/Parse -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources-normal/x86_64 -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources/x86_64 -I/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/DerivedSources -F/Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Products/Release-iphonesimulator/Parse -Wincomplete-umbrella -include /var/folders/95/xf9fwhxs1hx7ngv0mx4w9h9c0000gn/T/CocoaPods-Lint-20200329-37648-1vybrbm-Parse/Pods/Target\ Support\ Files/Parse/Parse-prefix.pch -MMD -MT dependencies -MF /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookAuthenticationProvider.d --serialize-diagnostics /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookAuthenticationProvider.dia -c /Users/drdaz/Documents/Development/Others/Parse-SDK-iOS-OSX/ParseFacebookUtils/ParseFacebookUtils/Internal/AuthenticationProvider/PFFacebookAuthenticationProvider.m -o /Users/drdaz/Library/Developer/Xcode/DerivedData/App-fynnfoaetgywzoastsfbzieyawga/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/Parse.build/Objects-normal/x86_64/PFFacebookAuthenticationProvider.o
    warning: Could not read serialized diagnostics file: Cannot Load File: Failed to open diagnostics file (in target 'Parse' from project 'Pods')
```

So the FBSDKCoreKit import appears to fail. Not sure why.

noobs2ninjas commented:
I removed all dynamic library links and re-added. Tested the building of each one and everything seems to build just fine now. Committed my changes.

noobs2ninjas commented:
Cocoapods could be an issue with versions on the Podspec. Haven't looked at that. Also, if I'm building with cocoapods then linking is different than with carthage. One we pull from the Carthage builds folder. The other would be included in the pods so we shouldnt be creating the dynamic from the cocoapods. That defeats the purpose of cocoapods.

drdaz commented:
"I removed all dynamic library links and re-added. Tested the building of each one and everything seems to build just fine now. Committed my changes.I removed all dynamic library links and re-added. Tested the building of each one and everything seems to build just fine now. Committed my changes."

The changes you pushed do more than you describe here - there's much more removed in this change than there is added. It looks like you also pruned some of the system's frameworks from the project. 

As far as I can tell the net result of these changes is that the cocoapods build error (FBSDKCoreKit module is not found when building ParseFacebookUtils) is now exposed in the main build log, rather than being buried under a meaningless 'something failed' message. And this is definitely a good thing.

But do how do we know which of the changes caused this result? I, at least, am not sure.

It's good practice to make small, controlled changes to the codebase that alter the system behaviour (and the test results) in understood and controlled ways. Otherwise fixing and learning become very, very hard. And in order to understand the effects of our changes, it's vital that we test locally and understand the effects of our work before we push.

As an example, the few changes I pushed to start this PR, I'd worked on for a day or 2 before pushing. I started trying a whole load of things. And when I got the result I was looking for, I started removing my changes until I found the smallest changeset that fixed the issue (Carthage build) before pushing.

drdaz commented:
As always, not going after you personally here, just trying to share what many years of making my own mistakes has taught me :)

noobs2ninjas commented:
You are correct about the additional lines removed. Previously those where causing warnings on every dynamic build target since you can’t build system specific frameworks into your target anymore. Pretty sure I still didn’t get them all, but what ones I did update removed unnecessary lines and remove some warnings. Didn’t think to note it because they shouldn’t have been linked to begin with.

noobs2ninjas commented:
In this case I’d didn’t see the need to do an incremental change since linking those system libraries has never worked to begin with. Only caused warnings.

drdaz commented:
Sure. And I thought about removing them myself when I fixed the Carthage build here; the build setting I changed makes manually linking system frameworks unnecessary.

It's better imho to have a separate 'cleanup' commit once things are tested and shown to be fixed. That way we can be more sure that the cleanup doesn't affect anything.  I don't know how many times I've tweaked something I was sure was harmless, unused, or isolated, only to then have to spend the next day or so fixing random broken stuff 😃

noobs2ninjas commented:
That probably still needs to happen. Figured I’d clean it all up for the targets that where experiencing linking issues. I think what issues we have left will be resolved with catching up the Podspec with the Carthage dependency versions.

drdaz commented:
Actually looking at the podspec, FBSDKCoreKit is just plain missing 😃. I've added it and I'm testing now.

noobs2ninjas commented:
at some point we also need to figure out why we are doing a cocoapods build on both Travis and circle. Seems a bit redundant.

noobs2ninjas commented:
Looking at the podspec wouldn’t listing all the system frameworks for each targets s.frameworks add the same warnings during the cocoapods build.

drdaz commented:
There don't seem to be any warnings related to those in the build... but I'll be honest I didn't notice the warnings they were producing from the links in the project.

FWIW the currently failing part of the test boils down to executing this: `pod lib lint Parse.podspec --allow-warnings --use-libraries`. Adding --verbose to the args provides a LOT more output.

noobs2ninjas commented:
Looking at our rake file. I see 
podspecs.each do |podspec|
      results << system("pod lib lint #{podspec} --allow-warnings")
      results << system("pod lib lint #{podspec} --allow-warnings --use-libraries")
    end

So seems we are building this twice one using libraries the other not. Is that intentional you think?

drdaz commented:
I do think it's intentional. I think one relates to our static targets while the other relates to dynamic. Currently the --using-libraries lint fails, while the other passes.

noobs2ninjas commented:
Maybe I messed this up. So I relinked everything based on where the Facebook core and login frameworks are located for Carthage. Those locations would not be the same for cocoapods. Perhaps we need to re-examine that.

drdaz commented:
I'm not sure I understand; could you dive deeper into that?

It doesn't seem to have affected the build negatively as of right now.

drdaz commented:
That said, the links to the FB libraries in the Vendor folder do need to be there:

<img width="1134" alt="Screenshot 2020-04-07 at 15 37 19" src="https://user-images.githubusercontent.com/845731/78675729-c10e3880-78e5-11ea-8a53-a4ae61c894e9.png">

drdaz commented:
... or maybe they don't. Perhaps adding ./Vendor to the framework search path does the same job. 🤷🏽‍♂️

But this does serve to confuse things (and possibly create new issues), since it's a change that may affect other things but doesn't seem to move us closer to the goal.

noobs2ninjas commented:
So in order to build dynamic targets for Carthage Dependencies. I linked the Facebook frameworks from Carthage/Build/iOS. Same for tvOS. The issue with this is cocoapods when using dynamic frameworks doesn’t output dependencies to that same folder.

noobs2ninjas commented:
Found an article that better explains the issue and a little bit about how to fix it. https://ppinera.es/2017/09/13/xcodembed/

noobs2ninjas commented:
Oh wait! Never mind. I thought this was a linking issue. The podspec just didn’t pass validation.

drdaz commented:
Yes 😃 But it fails validation because it can't compile; it can't find the FB dependency library.

noobs2ninjas commented:
Wait so this is happening with —use-frameworks set?

noobs2ninjas commented:
Ok I can’t take it anymore. I’m going to get my computer booted instead of looking through logs

drdaz commented:
> Wait so this is happening with —use-frameworks set?

Yes. And not without it. 🙂

noobs2ninjas commented:
Ok I figured it out. The new version of xcodebuild will build all targets in parallel making headers not available from FBSDKCore when FBSDKLoginKit is being built. Adding --use-modular-headers should fix it.

noobs2ninjas commented:
@drdaz  Check it out our [cocoapods test](https://app.circleci.com/pipelines/github/parse-community/Parse-SDK-iOS-OSX/37/workflows/cc028ea2-f718-4d10-9ae5-e8e8ceee1f71/jobs/4604) passed!

That one change fixed the whole thing. Now lets get this merged and well run the nightly build.

drdaz commented:
@noobs2ninjas That's awesome! Great find!

But why are the tests cancelled? Gotta see that they aren't broken by this change too (as unlikely as that may be).

noobs2ninjas commented:
Sorry I cancelled it. CircleCI just takes forever to run through all the tests and you cant prioritize them. Rest assured, it is literally impossible for my change to break anything else. Only two lines that have changed since last commit and now cocoapods works. 
```
Old:      results << system("pod lib lint #{podspec} --allow-warnings --use-libraries")
New:    results << system("pod lib lint #{podspec} --allow-warnings --use-libraries --use-modular-headers")
```

drdaz commented:
I manually re-ran the tests and they did as expected 😉.

But why oh why does the nightly cocoapods still seem to fail the same as it did before these changes??

drdaz commented:
Just ran the Carthage test target here... and it passed 😬🤷🏽‍♂️

drdaz commented:
I just realised. I didn't wake up to a mail about the nightly failing this morning! 🎉🍾🎊

noobs2ninjas commented:
That’s absolutely maddening. We changed absolutely nothing.

drdaz commented:
Heh... I guess that's another of our inconsistent tests.

noobs2ninjas commented:
Its been working for days now. So, as far as I'm concerned this is fixed. :) Well done!
